plonky2x/core/src/frontend/eth/rlp/decoder.rs

```diff
@@ -0,0 +1,249 @@
+//! This file implements RLP decoder.
+//!
+//! Reference: https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/
+//!
+//! We have a total of 6 different ways of representing Merkle Patricia Tries.
+//! - Encoded versions
+//!     1. Encoded MPT : vec<u8>
+//!     2. Padded Encoded MPT : ([u8; usize], usize)
+//!     3. Circuita Padded Encoded MPT : ArrayVariable<ByteVariable, ENCODING_LEN>
```

Comment:
Should this be a 2D array like 2?

Reply:
I added "Variable" that denotes the length, which I assume is what you meant by 2D array? (I did the same for `3'` as well.)